### PR TITLE
Issue152

### PR DESCRIPTION
**Pull request by jussi80**
Added variables, callbacks, events to Label to make it conform Rotatable protocol.

### CODE REVIEW

File: Sources/Core/Internal/TextLayer.swift [hunk 1]
```diff
@@ -8,7 +8,16 @@ import Foundation
...
+    var rotation: Metric = 0 { didSet { updateTransform() } }
     override func action(forKey event: String) -> CAAction? {
         return NSNull()
     }
+
+    // MARK: - Private
+
+    private func updateTransform() {
+        var newTransform = CATransform3DIdentity
+        newTransform = CATransform3DRotate(newTransform, rotation, 0, 0, 1)
```
Comment (JohnSundell) [inline Sources/Core/Internal/TextLayer.swift]: We can skip a mutation here by doing:
```
transform = CATransform3DMakeRotation(rotation, 0, 0, 1)
```

File: Tests/ImagineEngineTests/LabelTests.swift [hunk 2]
```diff
@@ -122,4 +122,21 @@ final class LabelTests: XCTestCase {
         XCTAssertEqual(labelBClickCount, 1)
         XCTAssertEqual(clickedLabels, [labelB, labelA, labelA])
     }
+    
+    func testObservingRotate() {
```
Comment (JohnSundell) [inline Tests/ImagineEngineTests/LabelTests.swift]: This is a nitpick, but how about naming this `testObservingRotation`?

File: Tests/ImagineEngineTests/LabelTests.swift [hunk 3]
```diff
@@ -122,4 +122,21 @@ final class LabelTests: XCTestCase {
...
+
+        var labelRotateCount = 0
+        label.events.rotated.observe {
+            labelRotateCount += 1
+        }
+
+        XCTAssertEqual(labelRotateCount, 0)
+        XCTAssertEqual(label.rotation, 0)
+        label.rotation = 90
+        XCTAssertEqual(label.rotation, 90)
```
Comment (JohnSundell) [inline Tests/ImagineEngineTests/LabelTests.swift]: I don't think you need to verify that the value of the property is the same that you just set, that makes the test harder to read IMO 🙂. Instead, I think you can focus on the relationship between changing `label.rotation` and `labelRotateCount`.

Comment (jussi80) [reply Tests/ImagineEngineTests/LabelTests.swift]: Old habit to make sure that getter & setter works properly. :) I'll fix your findings and make another commit soon.

**Comment by JohnSundell** [comment]
@jussi80 Code looks great 👍 But looks like merging #157 into master caused this to have conflicts. Could you resolve the conflicts and I'll merge this in? 🙂

**Comment by jussi80** [comment]
Done and done. 🙂

File: Sources/Core/API/Label.swift [hunk 4]
```diff
@@ -15,173 +15,185 @@ import QuartzCore
...
 
+    let objCString = NSString(string: text)
+
+    let rect = objCString.boundingRect(with: .zero,
+                                       options: [.usesLineFragmentOrigin],
+                                       attributes: [.font: font],
+                                       context: nil)
+    size = rect.size
+}
+
```
Comment (aranasaurus) [inline Sources/Core/API/Label.swift]: Looks like everything from here to the class declaration lost an indentation level and should be shifted to the right by 4 spaces. Other than that the code looks great!

**Comment by jussi80** [comment]
Now it's fixed. Should we use the Xcode's re-indent feature to make sure that source code have consistent structure?

**Comment by JohnSundell** [comment]
@jussi80 I think it's fine the way it is. We have SwiftLint, which is run locally as well when running the unit tests 🙂

**Pull request merged**